### PR TITLE
break error

### PR DESCRIPTION
**Pull request by zgfzgf**
#3952 
use goto

### CODE REVIEW

**Comment by zgfzgf** [comment]
when lotus-miner running error.
Crtl+C kill lotus-miner fail

add this fix
when lotus-miner running error.
Crtl+C kill lotus-miner success

2020-09-23T10:41:21.285+0800	INFO	storageminer	storage/miner.go:237	Computing WinningPoSt ;[{SealProof:0 SectorNumber:0 SealedCID:bagboea4b5abcbk5yoopylhy2z7e2lhzkvnaz5ztf2ze4izn2uho4ta4aij3vmddp}]; [143 187 164 24 34 51 93 224 123 92 109 239 181 107 179 172 156 163 251 215 194 99 123 199 227 244 76 255 153 8 255 51]
2020-09-23T10:41:21.288+0800	INFO	sectors	storage-sealing/states_sealing.go:448	Proving sector 1
2020-09-23T10:41:21.288+0800	WARN	advmgr	sector-storage/manager.go:468	ReleaseUnsealed todo
2020-09-23T10:41:25.867+0800	INFO	storageminer	storage/miner.go:244	GenerateWinningPoSt took 4.581137553s
2020-09-23T10:41:25.869+0800	INFO	miner	miner/miner.go:441	mined new block	{"cid": "bafy2bzacedlp23u5p2fg3suyx65zmbwfsfug5zyr7jzgvg7rcrfrezcapttc6", "height": "3679", "miner": "t01000", "parents": ["t01000"], "took": 4.594110893}
2020-09-23T10:41:25.870+0800	WARN	miner	miner/miner.go:443	CAUTION: block production took longer than the block delay. Your computer may not be fast enough to keep up	{"tMinerBaseInfo ": 0.003441505, "tDrand ": 0.000000109, "tPowercheck ": 0.000000096, "tTicket ": 0.006822136, "tSeed ": 0.000008684, "tPending ": 4.581984379, "tCreateBlock ": 0.001853984}
2020-09-23T10:41:25.870+0800	WARN	miner	miner/miner.go:263	mined block in the past	{"block-time": "2020-09-17T17:44:38.000+0800", "time": "2020-09-23T10:41:25.870+0800", "difference": 493007.87039219}
2020-09-23T10:41:25.870+0800	ERROR	miner	miner/miner.go:268	<!!> SLASH FILTER ERROR: produced block would trigger 'parent-grinding fault' consensus fault; miner: t01000; bh: bafy2bzacedlp23u5p2fg3suyx65zmbwfsfug5zyr7jzgvg7rcrfrezcapttc6, expected parent: bafy2bzacedcbf2gixnpniozr4owdtglodhyxthv4rytkxvciyr3yiliqyxqwc
2020-09-23T10:41:25.874+0800	WARN	miner	miner/miner.go:215	BestMiningCandidate from the previous round: [bafy2bzacedrf5ghiyhlopmbib5wrbyz4afddgysmqpslh2w37eyo5rzslk4m6] (nulls:0)
2020-09-23T10:41:29.881+0800	WARN	miner	miner/miner.go:215	BestMiningCandidate from the previous round: [bafy2bzacedrf5ghiyhlopmbib5wrbyz4afddgysmqpslh2w37eyo5rzslk4m6] (nulls:0)
2020-09-23T10:41:33.886+0800	WARN	miner	miner/miner.go:215	BestMiningCandidate from the previous round: [bafy2bzacedrf5ghiyhlopmbib5wrbyz4afddgysmqpslh2w37eyo5rzslk4m6] (nulls:0)
2020-09-23T10:41:37.895+0800	WARN	miner	miner/miner.go:215	BestMiningCandidate from the previous round: [bafy2bzacedrf5ghiyhlopmbib5wrbyz4afddgysmqpslh2w37eyo5rzslk4m6] (nulls:0)
^C2020-09-23T10:41:41.587+0800	WARN	main	lotus-storage-miner/run.go:167	received shutdown	{"signal": "interrupt"}
2020-09-23T10:41:41.587+0800	WARN	main	lotus-storage-miner/run.go:172	Shutting down...
2020-09-23T10:41:41.588+0800	INFO	miner	miner/miner.go:142	received interrupt while trying to sleep in mining cycle
^C
^C
^C

File: miner/miner.go [hunk 1]
```diff
@@ -151,6 +151,7 @@ func (m *Miner) mine(ctx context.Context) {
 	var lastBase MiningBase
 
 	for {
+	minerStop:
```
Comment (magik6k) [inline miner/miner.go:154]: We could change this to
```suggestion
minerLoop:
	for {
```

And then use `continue minerLoop` instead of goto below

**Review by magik6k — COMMENTED** [review]
Thanks for catching this. Just 1 nit

File: miner/miner.go (hunk 1 above)
Comment (zgfzgf) [reply miner/miner.go]: Now it's changed

**Pull request merged**